core/src/main/java/com/datastax/oss/driver/internal/core/metadata/MetadataManager.java

```diff
@@ -404,11 +415,7 @@ private void startSchemaRequest(CompletableFuture<Metadata> future) {
             .whenComplete(
                 (v, error) -> {
                   if (error != null) {
-                    Loggers.warnWithException(
-                        LOG,
-                        "[{}] Unexpected error while refreshing schema, skipping",
-                        logPrefix,
-                        error);
+                    currentSchemaRefresh.completeExceptionally(error);
                   }
                   singleThreaded.firstSchemaRefreshFuture.complete(null);
                 });
```

Comment:
In this line:
https://github.com/datastax/java-driver/blob/b3475942c80b83c379181516db9b158d6cc72407/core/src/main/java/com/datastax/oss/driver/internal/core/metadata/MetadataManager.java#L454
Are we still catching an exception and fail silently? Shouldn't we add the logging here?

Reply:
We are not failing silently, but propagating the error (via the future). Callers will either surface the future to the client, or add a callback that logs the error.